packages/snaps-jest/src/internals/simulation/interface.ts

```diff
@@ -496,32 +540,105 @@ export async function selectInDropdown(
 }
 
 /**
- * Get a user interface object from a Snap.
+ * Upload a file to an interface element.
  *
- * @param runSaga - A function to run a saga outside the usual Redux flow.
- * @param snapId - The Snap ID.
  * @param controllerMessenger - The controller messenger used to call actions.
- * @yields Takes the set interface action.
- * @returns The user interface object.
+ * @param id - The interface ID.
+ * @param content - The interface Components.
+ * @param snapId - The Snap ID.
+ * @param name - The element name.
+ * @param file - The file to upload. This can be a path to a file or a
+ * `Uint8Array` containing the file contents. If this is a path, the file is
+ * resolved relative to the current working directory.
+ * @param options - The file options.
+ * @param options.fileName - The name of the file. By default, this is
+ * inferred from the file path if it's a path, and defaults to an empty string
+ * if it's a `Uint8Array`.
+ * @param options.contentType - The content type of the file. By default, this
+ * is inferred from the file name if it's a path, and defaults to
+ * `application/octet-stream` if it's a `Uint8Array` or the content type
+ * cannot be inferred from the file name.
  */
-export function* getInterface(
-  runSaga: RunSagaFunction,
-  snapId: SnapId,
+export async function uploadFile(
   controllerMessenger: RootControllerMessenger,
-): SagaIterator<SnapInterface> {
-  const { type, id, content } = yield call(
-    getStoredInterface,
-    controllerMessenger,
+  id: string,
+  content: JSXElement,
+  snapId: SnapId,
+  name: string,
+  file: string | Uint8Array,
+  options?: FileOptions,
+) {
+  const result = getElement(content, name);
+
+  assert(
+    result !== undefined,
+    `Could not find an element in the interface with the name "${name}".`,
+  );
+
+  assert(
+    result.element.type === 'FileInput',
+    `Expected an element of type "FileInput", but found "${result.element.type}".`,
+  );
+
+  const { state, context } = controllerMessenger.call(
+    'SnapInterfaceController:getInterface',
     snapId,
+    id,
+  );
+
+  const fileObject = await getFileToUpload(file, options);
```

Comment:
We should reject files that are too large here, similar to how we will have to do in the client